hls4ml/templates/vivado_accelerator/alveo/python_drivers/axi_stream_driver.py

```diff
@@ -0,0 +1,109 @@
+from datetime import datetime
+
+import numpy as np
+from pynq import Overlay
+from pynq import allocate
+
+
+class NeuralNetworkOverlay(Overlay):
```

Comment:
I think we can hide a bit of the buffer handling from the user. For example, make `input_buffer` and `output_buffer` members of the class so the user doesn't have to pass them around between function calls. And `predict` could call `allocate_mem` itself.
It would be nice if the user code could be as simple as:
```
nn = NeuralNetworkOverlay('my.cxlbin')
y = nn.predict(X, y_shape)
```